### ISSUE TITLE
Blank Network alerts

### ISSUE THREAD

**Issue by Theo-user** [henrypp/simplewall#1009]
Latest Windows 10
2560x1440 resolution with no scaling
Version 3.4.3

![Capture](https://user-images.githubusercontent.com/75871295/131488022-696c2406-fa38-4eb4-8f2b-1c7d8a0bb348.PNG)

**Comment by VirusRenamon** [comment]
I have this issue with normal 1080p resolution, too. Dont think resolution has anything to do with this bug.

**Comment by Theo-user** [comment]
I believe it's a scaling issue. 

I went from 4k 175% scaling to 150% and to 100%
![Screenshot_20210901-212653_VNC Viewer](https://user-images.githubusercontent.com/75871295/131740370-db98fa6b-8e9d-4984-b599-1799188d2540.jpg)
![Screenshot_20210901-212739_VNC Viewer](https://user-images.githubusercontent.com/75871295/131740377-c73289fa-e950-4084-892a-0d4a680fab4e.jpg)

**Comment by eric15342335** [comment]
I can reproduce this error.

**Comment by Chaython** [comment]
I had not experienced this issue until 3.4.3

**Comment by Theo-user** [comment]
> I had not experienced this issue until 3.4.3

The images are with 3.4.3 so not fixed here.

**Comment by Chaython** [comment]
> 
> 
> > I had not experienced this issue until 3.4.3
> 
> The images are with 3.4.3 so not fixed here.

As I said "until" so use an older version, the bug didn't exist.

**Comment by Theo-user** [comment]
> > > I had not experienced this issue until 3.4.3
> > 
> > 
> > The images are with 3.4.3 so not fixed here.
> 
> As I said "until" so use an older version, the bug didn't exist.

I also had it with 3.4.2

**Comment by Matthew-afk-gif** [comment]
3.4.3 still

**Comment by Matthew-afk-gif** [comment]
turn off the second monitor (pull out the VGA or HDMI cable ), turn on the simplewall, turn on the second monitor (plug in) and you will be ready to use simplewall with 2 monitors

**Comment by Theo-user** [comment]
> turn off the second monitor (pull out the VGA or HDMI cable ), turn on the simplewall, turn on the second monitor (plug in) and you will be ready to use simplewall with 2 monitors

I have this issue with a single monitor system also

**Comment by eric15342335** [comment]
maybe the "network alert" is not loaded on system startup?

**Comment by bfgh012** [comment]
I'm having this issue after updating to 3.5.1
And it's not just in Network alerts, but in other parts of the UI as well, like the context menus.
![Screenshot 2021-10-16 015927](https://user-images.githubusercontent.com/47962561/137574515-aa7c49ff-1f96-46b3-9a56-2b251cd14250.png)
![Screenshot 2021-10-16 020025](https://user-images.githubusercontent.com/47962561/137574517-5f7546df-31d4-40e8-8527-937b60e7bdf3.png)

**Comment by henrypp** [comment]
@bfgh012 i see this, this is fixed, release soon.

**Comment by henrypp** [comment]
@bfgh012 

v3.5.3 fixed this

**Comment by Theo-user** [comment]
![Capture](https://user-images.githubusercontent.com/75871295/138379799-90b17027-e9e9-4259-a207-0874d74cfc9d.PNG)

**Comment by henrypp** [comment]
@Theo-user @cronyakatsuki

Try this one, please.

[simplewall-3.5.3.3-bin.zip](https://github.com/henrypp/simplewall/files/7449630/simplewall-3.5.3.3-bin.zip)

**Comment by Theo-user** [comment]
> @Theo-user @cronyakatsuki
> 
> Try this one, please.
> 
> [simplewall-3.5.3.3-bin.zip](https://github.com/henrypp/simplewall/files/7449630/simplewall-3.5.3.3-bin.zip)
> 
> ps: i still cannot catch this issue on my computer, dpi changing on all monitors does not get this error. (19044)

I've just given this a quick test on one of my systems,  the one least affected but reproducible there.  I think you could have fixed it. I will continue testing it.  Thanks mate.

**Comment by Theo-user** [comment]
> @Theo-user @cronyakatsuki
> 
> Try this one, please.
> 
> [simplewall-3.5.3.3-bin.zip](https://github.com/henrypp/simplewall/files/7449630/simplewall-3.5.3.3-bin.zip)
> 
> ps: i still cannot catch this issue on my computer, dpi changing on all monitors does not get this error. (19044)

It is still happening here. A few sleep/wake cycles and it's back.

**Comment by FritesMoules** [comment]
So far it's fixed for me  :))

**Comment by Theo-user** [comment]
Yeah definitely not fixed here. Here is another issue that's most likely related.

3.5.3.3

![Capture](https://user-images.githubusercontent.com/75871295/139670951-62d7465f-4370-4b8a-b165-0f30cf6a38e9.PNG)

**Comment by henrypp** [comment]
@Theo-user

> Yeah definitely not fixed here. Here is another issue that's most likely related.

this is fixed.

You can download test build from here:
https://www.henrypp.org/files/simplewall/latest/

**Comment by Theo-user** [comment]
> 
> 
> @Theo-user
> 
> > Yeah definitely not fixed here. Here is another issue that's most likely related.
> 
> this is fixed.
> 
> You can download test build from here: https://www.henrypp.org/files/simplewall/latest/

Windows 11 now. 

This is a new one, slightly different. 

This is with 3.5.3.4


![Capture](https://user-images.githubusercontent.com/75871295/139951471-831ed554-f8db-4e1f-af70-f94a636e322b.PNG)

![Capture2](https://user-images.githubusercontent.com/75871295/139951522-0bc39c6c-357b-4ac6-8c0a-e199e3a71ea2.PNG)

**Comment by henrypp** [comment]
Try new build.
https://www.henrypp.org/files/simplewall/latest/

**Comment by Theo-user** [comment]
> Try new build. https://www.henrypp.org/files/simplewall/latest/

All I did was close the previous and install the latest. Deleted a rule so it would popup, and again, blank. I'm starting to have nightmares now with blank network alerts.

**Comment by bfgh012** [comment]
I see the issue still drags on...
In my case, upgrading to v3.5.3 completely resolved the issue.
If I can help in any way, I'm available.

**Comment by Powsey** [comment]
Getting some blank alerts as well
- Simplewall version: 3.6.1
- OS: Windows 11 - Version 10.0.22000 Build 22000

![image](https://user-images.githubusercontent.com/77591092/142069627-7fef1097-3a3b-4298-9f44-69f4fc175841.png)

**Comment by rbreaves** [comment]
My DPI is set to 225% and I am having this issue on a 2560x1440 monitor. This has worked around the issue on 3.6.1 @henrypp 

![image](https://user-images.githubusercontent.com/10969616/142742797-46af18de-3ab2-4941-b1f9-3523bfa5ecb1.png)

**Comment by ak-42** [comment]
I get this as well (3.6.1). I also have a multi monitor set-up, with different scaling per monitor, and I switch often with a KVM switch. I also never saw this on Win10 but it's happening pretty reliably on Win11.<br>Also, exiting the program from the tray and starting it up again is a pretty reliable workaround for me.

**Comment by Theo-user** [comment]
> My DPI is set to 225% and I am having this issue on a 2560x1440 monitor. This has worked around the issue on 3.6.1 @henrypp
> 
> ![image](https://user-images.githubusercontent.com/10969616/142742797-46af18de-3ab2-4941-b1f9-3523bfa5ecb1.png)

I haven't had the issue since doing this so fingers crossed. Thanks.

**Comment by rbreaves** [comment]
> I also never saw this on Win10 but it's happening pretty reliably on Win11.

@ak-42 Same here - this may very well be a thing specific to those on Windows 11. (not the insider track, but the normal release updates). And yea since overriding the dpi to scale by System it seems to have fixed the issue. Exiting and reopening the program was never a workable work around for me personally.

**Comment by khrunks** [comment]
I also get this intermitently.  3.6.1 release 64 bit.  New user no past history with other versions.  On windows 10.  Tried scaling 150% and 125%, does not matter.  On some boots it is constantly there and on others it is consistently abcent.

![sw-alert](https://user-images.githubusercontent.com/95271973/143996877-59df47eb-7722-495c-bc5b-4a8cc64cf4ef.jpg)
%

**Comment by Theo-user** [comment]
> I also get this intermitently. 3.6.1 release 64 bit. New user no past history with other versions. On windows 10. Tried scaling 150% and 125%, does not matter. On some boots it is constantly there and on others it is consistently abcent.
> 
> ![sw-alert](https://user-images.githubusercontent.com/95271973/143996877-59df47eb-7722-495c-bc5b-4a8cc64cf4ef.jpg) %

Try the fix above. I haven't had the issue since doing it.

**Comment by Theo-user** [comment]
Ok nevermind. The issue popped up again even with the setting above.

**Comment by DukeDukeEm** [comment]
![image](https://user-images.githubusercontent.com/69652463/147415105-6e16c825-cb7d-49e7-bed0-0b641861ab50.png)

**Comment by jacob-44** [comment]
Unfortunately I'm experiencing the same issue. Blank pop-up window on 4k screen. Unable to decide and take action.

**Comment by qupear** [comment]
![image](https://user-images.githubusercontent.com/8069685/147909366-837d34aa-e52a-4f6f-8124-67abc739ed4c.png)
In my case, this issue is fixed when I set scaling to 100%. I have two displays, and this issue can be easily reproduced by setting up to 125% scaling for one of the displays. There should be some check-in code to scale simplewall windows.

**Comment by Nura57** [comment]
Please is there a fix for windows 11?

**Comment by Powsey** [comment]
For me I killed simplewall and restarted it. It did the trick for me

Kind regards,
Cyrille Dagregorio

**Comment by jacob-44** [comment]
> For me I killed simplewall and restarted it. It did the trick

Only once, or do you have to repeat this?

**Comment by Powsey** [comment]
For me I haven’t had the blank window again. So hopefully once or once in a while if it happens again

**Comment by Nura57** [comment]
> > For me I killed simplewall and restarted it. It did the trick
> 
> Only once, or do you have to repeat this?

for me It's every time, it keeps happening even after a restart, that was why I was asking if there was any fixes.

**Comment by rbreaves** [comment]
I know this only worked for Theo temporarily, but have any of you guys tried what I suggested as a fix? Works for me just fine under windows 11.
https://github.com/henrypp/simplewall/issues/1009#issuecomment-974721400

**Comment by wolfchan33** [comment]
same issue here. suggested solution does not fix.

**Comment by Nura57** [comment]
> 

I just went back to version 3.5.3  Everything is working fine now.

**Comment by Theo-user** [comment]
> > 
> 
> I just went back to version 3.5.3 Everything is working fine now.

For now, sadly.

**Comment by Powsey** [comment]
I must admit I haven’t had this issue for a long time now. I run on the latest version.

Can you give maybe more details about your environment and the dump generated by the software? This should help Henry

**Comment by wolfchan33** [comment]
I had the same problem with 3.5.3 too

**Comment by Nura57** [comment]
> I must admit I haven’t had this issue for a long time now. I run on the latest version. Can you give maybe more details about your environment and the dump generated by the software? This should help Henry

@Powsey How do we do that?

**Comment by brianaviles** [comment]
I have had this issue since I started using this program about 6 months ago, and currently on 3.6.1. My display is 5K (5120x2880) and the only thing that briefly fixes this is closing the app and relaunching it. Eventually, once my computer goes into sleep or if I play a game, I'll get another blank alert.
![image](https://user-images.githubusercontent.com/8651473/152473569-b116a13c-f4a2-4cdf-b3d2-156e4d062852.png)

**Comment by Powsey** [comment]
> > I must admit I haven’t had this issue for a long time now. I run on the latest version. Can you give maybe more details about your environment and the dump generated by the software? This should help Henry
> 
> @Powsey How do we do that?

You should enable logging to file (1 on screenshot) then when the issue occurs, press 2 to retrieve the log file

![image](https://user-images.githubusercontent.com/77591092/152570752-9af8076b-1d40-4c72-ae96-3d1ea2ecf432.png)

**Comment by Powsey** [comment]
Erf, just had the issue again as well right now. I need to try to get some logging info as well.
Bug is still there indeed.

**Comment by Nura57** [comment]
> > > I must admit I haven’t had this issue for a long time now. I run on the latest version. Can you give maybe more details about your environment and the dump generated by the software? This should help Henry
> > 
> > 
> > @Powsey How do we do that?
> 
> You should enable logging to file (1 on screenshot) then when the issue occurs, press 2 to retrieve the log file
> 
> ![image](https://user-images.githubusercontent.com/77591092/152570752-9af8076b-1d40-4c72-ae96-3d1ea2ecf432.png)
[simplewall.log](https://github.com/henrypp/simplewall/files/8008003/simplewall.log)
I started it a day before it happened and I stopped immediately after clicking close on the blank popup.

**Comment by Powsey** [comment]
Yep same for me at 13:29 something.
I hope this will help @henrypp 
[Simple wall.log](https://github.com/henrypp/simplewall/files/8008018/Simple.wall.log)

At the moment the blank window is on a loop for me. I'm going to close and restart simplewall. Last time it did the trick for a while.

**Comment by jacob-44** [comment]
Also still experiencing this issue frequently but randomly.

**Comment by brianaviles** [comment]
Just got another blank alert. Checked the logs and all I see is info on the connection that was blocked, but nothing on the status of the window.

**Comment by Theo-user** [comment]
> I have a fix for this when it happens which is minimize the window, then bring it up again. The issue is that you can't do that natively on Windows as the notification only has the X to close it. I use Displayfusion so I have minimize and minimize to tray for every window

Thanks for the workaround

**Comment by henrypp** [comment]
check 3.6.2

**Comment by Theo-user** [comment]
> 

Good news. I've done all the usual tests to reproduce it and I am not able to, so it looks like it's fixed. Thank you.

**Comment by Theo-user** [comment]
It has happened again. PC went to sleep earlier today and now I get this.

![Screenshot 2022-04-26 202707](https://user-images.githubusercontent.com/75871295/165377101-1df123f0-941d-435f-9d5d-1aa8b321f46c.png)

**Comment by brianaviles** [comment]
definitely not fixed. had this issue last night while playing COD. only workaround i've found is closing the app and restarting.

**Comment by brianaviles** [comment]
seems to me like this must be a scaling issue. had this happen again last night while launching fortnite. will try to reproduce and take a screen recording

**Comment by qupear** [comment]
Yes, this bug is still active in 3.6.2. After installation everything was working fine but after a couple of sleep/wake up cycles simplewall now shows blank alerts. I suppose I need to close it and all will be fine again but I was waiting if this bug will appear and it does.

**Comment by akoppa** [comment]
This bug is chronic or maybe each Windows update messes up the whole thing.

**Comment by hornyjailfish** [comment]
have this issue too

**Comment by Theo-user** [comment]
> This bug is chronic or maybe each Windows update messes up the whole thing.

I don't look at the code but the coordinates used to draw the controls onto the network alerts window is getting skewed. They are being drawn, just not where they should be.
I'm not sure exactly what it is causing it but it's either scaling/resolution/monitor count, or even simply a monitor sleep, that can trigger it. Windows is buggy with this kind of thing. Whatever Simplewall uses to figure out where to place the controls needs to be changed/improved.

**Comment by Bohan-Song** [comment]
**The same problem was solved using this method**
_1080P dual monitor Win10  DPI：100%_
![Snipaste_2022-05-10_15-44-29](https://user-images.githubusercontent.com/15974667/167576285-afe8c1b4-19aa-4f8c-ac3c-af41ed28caac.png)

**Comment by Theo-user** [comment]
> **The same problem was solved using this method** _1080P dual monitor Win10 DPI：100%_ ![Snipaste_2022-05-10_15-44-29](https://user-images.githubusercontent.com/15974667/167576285-afe8c1b4-19aa-4f8c-ac3c-af41ed28caac.png)

English?

**Comment by hornyjailfish** [comment]
thanks this is working solution

**Comment by henrypp** [comment]
Attempt n.9999. Try 3.6.3.

**Comment by Lod3** [comment]
Still get blank network alerts, it usually happens after system "wakes" up. 
After restart of simplewall the network alerts are not blank anymore. 
![simplewall_ax1lTyfupH](https://user-images.githubusercontent.com/14292591/167818330-bc141c0f-0ede-495a-85b3-bb41789ac58d.png)

**Comment by rbreaves** [comment]
Not sure if henrypp is playing whackamole or what but I am beginning to feel bad for you 😅.

**Comment by Theo-user** [comment]
It's not fixed but I guess you could say it's better as I can still interact with the window.

![Screenshot 2022-05-14 192308](https://user-images.githubusercontent.com/75871295/168443949-14485038-5945-4de3-911f-56559c7acc88.png)

**Comment by Chaython** [comment]
I still get blanks too, it's only after waking from sleep, or turning a monitor on/off.
Has to do with the mixed resolutions / scaling

**Comment by Kevin-FR** [comment]
<img width="1018" alt="SimpleWall" src="https://user-images.githubusercontent.com/82709307/168469061-561948aa-214a-4ea5-9a11-83cd11d35a83.png">
Same

**Comment by Nura57** [comment]
I've  been using 3.6.2 for a while now and have yet to come across the issue. It has been prompting me to update but I'm scared to do so as it might break it again. I'll update if it happens.

**Comment by qupear** [comment]
It might sound stupid, but to solve this issue it may be enough to check if another display was connected, then restart simplewall or if PC was wake up from sleep or hibernation then too restart simplewall.

**Comment by henrypp** [comment]
@Theo-user 

try this build, i removed shitty by design PerMonitor (thx m$) option (only PerMonitorV2 was left).

[simplewall.zip](https://github.com/henrypp/simplewall/files/8777354/simplewall.zip)

**Comment by Theo-user** [comment]
> @Theo-user
> 
> try this build, i removed shitty by design PerMonitor (thx m$) option (only PerMonitorV2 was left).
> 
> [simplewall.zip](https://github.com/henrypp/simplewall/files/8777354/simplewall.zip)

A few system sleep/wake cycles and the issue came back with this build too.

**Comment by brianaviles** [comment]
just got this alert on 3.6.3

![Screenshot 2022-06-13 093638](https://user-images.githubusercontent.com/8651473/173402365-07e59dea-f928-452d-9d77-d4ad0a22f577.png)

**Comment by Braboware** [comment]
I had the blank network alerts issue with 3.6.2 and 3.6.3, but it was resolved when I left localization ticked on during reinstallation of 3.6.3. Worth a shot if the issue persists and anyone ticked it off during installation

**Comment by henrypp** [comment]
I think it fixed by now.

**Comment by rbreaves** [comment]
> I still get blanks too, it's only after waking from sleep, or turning a monitor on/off.
> Has to do with the mixed resolutions / scaling

Oddly enough it never occurs over RDP for me lol.. and I am changing my resolution and scaling quite often, but I am never sleeping or turning anything off monitor wise.

**Comment by Lod3** [comment]
Sorry man :/ I updated today, I still get blank notification 
![chrome_yp8qYujRdE](https://user-images.githubusercontent.com/14292591/174575457-ed207195-67f0-41d2-b42c-f661d78e6027.png)
![simplewall_jtZxHju0dD](https://user-images.githubusercontent.com/14292591/174575746-36476c6b-268c-4f85-a45f-53e39f4b16a3.png)

**Comment by Chaython** [comment]
It still always breaks after waking from sleep, when I have one monitor that's a different resolution / scaling factor than the others.

**Comment by rbreaves** [comment]
https://www.youtube.com/watch?v=qYS0EeaAUMw

**Comment by henrypp** [comment]
@Theo-user @Lod3 @Chaython @x33st0rm

Hi guys, i think i finally fixed this issue. Try this v3.6.4.1 pre-release build here:
https://www.henrypp.org/files/simplewall/latest/

I have not enough time for now for testing this, but i finally fixed invalid dpi values and sizing on hidden notification window.

There is can be some bugs in UI by now, but in final version i fix them.

**Comment by Theo-user** [comment]
It's been a while but I have yet to see this issue crop up again. I believe you fixed it this time. Good job mate.

**Comment by btmccord** [comment]
I am having this same issue on the 3.8.5 release

![Screenshot 2024-12-02 201838](https://github.com/user-attachments/assets/584c2c79-6122-40b3-98ed-32530dfa3307)
